docs/side_quests/nf-test.md

```diff
@@ -0,0 +1,1074 @@
+
+### 2.1. Test the `sayHello` process
+
+Let's start with the `sayHello` process.
+
+Let's use the `nf-test generate` command again to generate tests for the process.
+
+```bash
+nf-test generate process main.nf
+```
+
+```console title="Output"
+> nf-test generate process main.nf
+
+Load source file '/workspaces/training/side-quests/nf-test/main.nf'
+Wrote process test file '/workspaces/training/side-quests/nf-test/tests/main.sayhello.nf.test
+Wrote process test file '/workspaces/training/side-quests/nf-test/tests/main.converttoupper.nf.test
+
+SUCCESS: Generated 2 test files.
+```
+
+Let's focus for now and focus on the `sayhello` process in the `main.sayhello.nf.test` file.
+
+Let's open the file and take a look at the contents.
+
+```groovy title="tests/main.sayhello.nf.test"
+nextflow_process {
+
+    name "Test Process sayHello"
+    script "main.nf"
+    process "sayHello"
+
+    test("Should run without failures") {
+
+        when {
+            params {
+                // define parameters here. Example:
+                // outdir = "tests/results"
+            }
+            process {
+                """
+                // define inputs of the process here. Example:
+                // input[0] = file("test-file.txt")
+                """
+            }
+        }
+
+        then {
+            assert process.success
+            assert snapshot(process.out).match()
+        }
+
+    }
+
+}
+```
+
+As before, we start with the test details, followed by the `when` and `then` blocks. However, we also have an additional `process` block which allows us to define the inputs to the process.
+
+Let's run the test to see if it works.
+
+```bash title="nf-test pipeline pass"
+nf-test test tests/main.sayhello.nf.test
+```
+
+```console title="Process test fails"
+> nf-test test tests/main.sayhello.nf.test
+
+🚀 nf-test 0.9.2
+https://www.nf-test.com
+(c) 2021 - 2024 Lukas Forer and Sebastian Schoenherr
+
+
+Test Process sayHello
+
+  Test [1eaad118] 'Should run without failures' FAILED (4.876s)
+
+  Assertion failed:
+
+  assert process.success
+         |       |
+         |       false
+         sayHello
+
+  Nextflow stdout:
+
+  Process `sayHello` declares 1 input channel but 0 were specified
+
+   -- Check script '/workspaces/training/side-quests/nf-test/.nf-test-1eaad118145a1fd798cb07e7dd75d087.nf' at line: 38 or see '/workspaces/training/side-quests/nf-test/.nf-test/tests/1eaad118145a1fd798cb07e7dd75d087/meta/nextflow.log' file for more details
+  Nextflow stderr:
+
+  Nextflow 24.10.4 is available - Please consider updating your version to it
+
+
+
+
+FAILURE: Executed 1 tests in 4.884s (1 failed)
+```
+
+The test fails because the `sayHello` process declares 1 input channel but 0 were specified. Let's fix that by adding an input to the process. Remember from part 1, our `sayHello` process takes a single value input. We should also fix the test name to better reflect what we're testing.
```

Comment:
```suggestion
The test fails because the `sayHello` process declares 1 input channel but 0 were specified. Let's fix that by adding an input to the process. Remember from Part 1, our `sayHello` process takes a single value input. We should also fix the test name to better reflect what we're testing.
```

We've been capitalizing `Part` in the Hello Nextflow training, so suggesting this here for consistency.

Reply:
Thanks for highlighting. Actually, referencing part 1 isn't very helpful here, since we're on a side-quest. I probably need to re-introduce the workflow content quickly early on